packages/thirdweb/src/react/core/hooks/others/useProfiles.ts

```diff
@@ -0,0 +1,36 @@
+import { type UseQueryResult, useQuery } from "@tanstack/react-query";
+import type { Profile } from "../../../../wallets/in-app/core/authentication/types.js";
+import { useActiveWallet } from "../wallets/useActiveWallet.js";
+
+/**
+ * @description Retrieves all linked profiles for the current wallet.
+ *
+ * @returns A React Query result containing the linked profiles for the given wallet.
+ *
+ * @example
+ * ```jsx
+ * import { use } from "thirdweb/react";
+ *
+ * const { data: profiles } = useProfiles();
+ *
+ * console.log("Type:", profiles[0].type); // "discord"
+ * console.log("Email:", profiles[0].email); // "john.doe@example.com"
+ * ```
+ *
+ * @wallet
+ */
+export function useProfiles(): UseQueryResult<Profile[]> {
+  const wallet = useActiveWallet();
+
+  return useQuery({
+    queryKey: ["profiles"],
```

Comment:
query key **has** to include wallet identifier because the profiles depend on which wallet is connected, in this case likely you need the wallet id + the address (because I assume both of those play a part in determining what data gets returned)